client/asset/eth/eth.go

```diff
@@ -5238,34 +5243,70 @@ func (w *ETHWallet) WalletTransaction(ctx context.Context, txID string) (*asset.
 	return w.getReceivingTransaction(ctx, txHash)
 }
 
-func (w *TokenWallet) getReceivingTransaction(ctx context.Context, txHash common.Hash) (*asset.WalletTransaction, error) {
-	tx, blockHeight, err := w.node.getTransaction(ctx, txHash)
+// extractValueFromTransferLog checks the Transfer event logs in the
+// transaction, finds the log that sends tokens to the wallet's address,
+// and returns the value of the transfer.
+func (w *TokenWallet) extractValueFromTransferLog(receipt *types.Receipt) (uint64, error) {
+	if len(receipt.Logs) == 0 {
+		return 0, fmt.Errorf("no logs found in receipt")
+	}
+
+	tokenContract, err := erc20.NewIERC20(w.netToken.Address, w.node.contractBackend())
 	if err != nil {
-		return nil, err
+		return 0, fmt.Errorf("NewIERC20 error: %v", err)
 	}
-	if *tx.To() != w.netToken.Address {
-		return nil, asset.CoinNotFoundError
+
+	var transferredAmt uint64
+	for _, log := range receipt.Logs {
+		if log.Address != w.netToken.Address {
```

Comment:
Let's keep this kind of logic in the `tokenContractor`. https://github.com/buck54321/dcrdex/compare/3054b63...buck54321:dcrdex:contractor-parse-transfer